pkgs/development/ocaml-modules/happy-eyeballs/mirage.nix

```
@@ -0,0 +1,42 @@
+{ lib, buildDunePackage
+, happy-eyeballs
+, duration
+, dns-client
+, domain-name
+, ipaddr
+, fmt
+, logs
+, lwt
+, mirage-clock
+, mirage-random
+, mirage-time
+, tcpip
+}:
+
+buildDunePackage {
+  pname = "happy-eyeballs-mirage";
+
+  inherit (happy-eyeballs) src version;
+
+  minimalOCamlVersion = "4.08";
+
+  strictDeps = true;
+
+  propagatedBuildInputs = [
+    happy-eyeballs
+    duration
+    dns-client
+    domain-name
+    ipaddr
+    fmt
+    logs
```

```suggestion
    dns-client
```

`duration domain-name ipaddr fmt logs` is to be propagated by `happy-eyeballs`